prov/rxm/src/rxm_conn.c

```diff
@@ -1303,7 +1303,7 @@ static ssize_t rxm_eq_sread(struct rxm_ep *rxm_ep, size_t len,
 	} else {
 		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "Unknown error: %d\n",
 			entry->err_entry.err);
-		return rd;
+		return -entry->err_entry.err;
```

Comment:
`rxm_conn_handle_event` would process error only if rd was set to `-FI_ECONNREFUSED`. In the else case `rd` should be set to `-FI_EAVAIL`.

Reply:
The CM thread will exit if the rxm_eq_sread() return is < 0 and not equal to "-FI_ECONNREFUSED". I'm not following why it is desirable in the else clause to *not* return the the actual error since the error has already been retrieved from the underlying message EQ. However, I'm fine to change to -FI_EAVAIL or at least set it to -FI_EOTHER if the err_entry.err is non-zero, if that is what you really prefer. I don't believe it will change the code path.

Note: rxm_conn_handle_event() assumes that it is passed a good event if the entry->rd != FI_CONNREFUSED because "FI_NOTIFY" is defined to to be 0 and the event->type is left at its initialized value of 0. If we decide to pass the error to the connection handler, this will need to be addressed and/or an else clause added to handle the other errors.

Reply:
Looks to me like "rd" will equal the size of struct fi_eq_err_entry (the return of reading the error entry from the message provider which is done by the RXM_EQ_READERR macro).

Reply:
Ah, I missed that. Thanks for checking again!

This area should probably be refactored to be more explicit about what's going on.

Reply:
That is a good point, I hate when macros update values like that. In the atomic auto updates I do reconfigure this logic some. I could eliminate the macro all together in those changes, as it is only used in the one new routine that reads the EQ error.

Reply:
This macro isn't used in other places. Feel free to remove it if you think that's better. :)